ethereum/eip712/encoding.go

```diff
@@ -0,0 +1,217 @@
+package eip712
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+
+	cosmosTypes "github.com/cosmos/cosmos-sdk/types"
+	txTypes "github.com/cosmos/cosmos-sdk/types/tx"
+
+	apitypes "github.com/ethereum/go-ethereum/signer/core/apitypes"
+	ethermint "github.com/evmos/ethermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+var ethermintProtoCodec codec.ProtoCodecMarshaler
+var ethermintAminoCodec *codec.LegacyAmino
+
+// The process of unmarshaling SignDoc bytes into a SignDoc object requires having a codec
+// populated with all relevant message types. As a result, we must call this method on app
+// initialization with the app's encoding config.
+func SetEncodingConfig(cfg params.EncodingConfig) {
+	ethermintAminoCodec = cfg.Amino
+	ethermintProtoCodec = codec.NewProtoCodec(cfg.InterfaceRegistry)
+}
+
+// Get the EIP-712 object hash for the given SignDoc bytes by first decoding the bytes into
+// an EIP-712 object, then hashing the EIP-712 object to create the bytes to be signed.
+// See https://eips.ethereum.org/EIPS/eip-712 for more.
+func GetEIP712HashForMsg(signDocBytes []byte) ([]byte, error) {
+	var typedData apitypes.TypedData
+
+	// Attempt to decode as both Amino and Protobuf since the message format is unknown.
+	// If either decode works, we can move forward with the corresponding typed data.
+	typedDataAmino, errAmino := decodeAminoSignDoc(signDocBytes)
+	typedDataProtobuf, errProtobuf := decodeProtobufSignDoc(signDocBytes)
+
+	switch {
+	case errAmino == nil:
+		typedData = typedDataAmino
+	case errProtobuf == nil:
+		typedData = typedDataProtobuf
+	default:
+		return make([]byte, 0), fmt.Errorf("could not decode sign doc as either Amino or Protobuf.\n amino: %v\n protobuf: %v\n", errAmino, errProtobuf)
+	}
+
+	domainSeparator, err := typedData.HashStruct("EIP712Domain", typedData.Domain.Map())
+	if err != nil {
+		return nil, fmt.Errorf("could not hash EIP-712 domain: %w", err)
+	}
+	typedDataHash, err := typedData.HashStruct(typedData.PrimaryType, typedData.Message)
+
+	if err != nil {
```

Comment:
```suggestion

	typedDataHash, err := typedData.HashStruct(typedData.PrimaryType, typedData.Message)
	if err != nil {
```

Reply:
Not following, what's the change here?

Reply:
@austinchandra More likely just a blank line deletion